search/tasks.py

```diff
@@ -63,19 +63,34 @@ def update_document_with_partial(doc_id, partial_data, object_type):
 
 
 @app.task(**PARTIAL_UPDATE_TASK_SETTINGS)
-def increment_document_integer_field(doc_id, field_name, incr_amount, object_types):
+def increment_document_integer_field(doc_id, field_name, incr_amount, object_type):
     """Task that makes a request to increment some integer field in an ES document"""
-    api.increment_document_integer_field(doc_id, field_name, incr_amount, object_types)
+    api.increment_document_integer_field(doc_id, field_name, incr_amount, object_type)
 
 
 @app.task
-def update_field_values_by_query(query, field_name, field_value, object_types):
+def update_field_values_by_query(query, field_dict, object_types):
     """
     Task that makes a request to update a field value for all ES documents that match some query.
     """
-    return api.update_field_values_by_query(
-        query, field_name, field_value, object_types
-    )
+    return api.update_field_values_by_query(query, field_dict, object_types)
+
+
+@app.task(autoretry_for=(RetryException,), retry_backoff=True, rate_limit="600/m")
```

Comment:
Since we're not contacting reddit when indexing profiles, I don't think we need this decorator or any of the rate limit or exponential backoff logic.

Reply:
✅